jobs/periodic/anonymise-attendance-log/index.ts

```diff
@@ -0,0 +1,18 @@
+import { getLogger } from "log4js";
+import { getManager } from "typeorm";
+import { CourseAttendanceLog } from "../../../common/entity/CourseAttendanceLog";
+
+const DELETE_AFTER = 7; // in days
+const logger = getLogger();
+
+export default async function execute() {
+    const entityManager = getManager();
+    await entityManager
+        .createQueryBuilder(CourseAttendanceLog, 'courseAttendanceLog')
+        .update()
+        .where('courseAttendanceLog.createdAt < :date', { date: Date.now() - DELETE_AFTER })
+        .set({ ip: "", pupil: { id: 0 } })
```

Comment:
So here the problem is that the pupilId is non nullable?

Reply:
So what pupilId should I then reference?
![image](https://user-images.githubusercontent.com/48623649/206905481-eca3961c-46e3-4e69-a321-1247bd48da85.png)

Should it point to an existing empty Mustermann pupil?

Reply:
Yes, that's fine. There is no need to anonymize the (internal) pupilId. As long as no IP address, names or email addresses are left if a user deactivates their account it should be okay 👍